go/libraries/doltcore/sqle/reflog_table_function.go

```diff
@@ -204,20 +195,28 @@ func (rltf *ReflogTableFunction) IsReadOnly() bool {
 }
 
 func (rltf *ReflogTableFunction) Expressions() []sql.Expression {
-	if rltf.refExpr != nil {
-		return []sql.Expression{rltf.refExpr}
-	}
 	return []sql.Expression{}
 }
 
 func (rltf *ReflogTableFunction) WithExpressions(expression ...sql.Expression) (sql.Node, error) {
-	if len(expression) > 1 {
-		return nil, sql.ErrInvalidArgumentNumber.New(rltf.Name(), "0 or 1", len(expression))
+	if len(expression) > 2 {
+		return nil, sql.ErrInvalidArgumentNumber.New(rltf.Name(), "0 to 2", len(expression))
 	}
 
 	new := *rltf
-	if len(expression) > 0 {
-		new.refExpr = expression[0]
+	args, err := getDoltArgs(rltf.ctx, expression, rltf.Name())
```

Comment:
A similar concern here with expressions... `getDoltArgs` is going to evaluate the expression, but we don't know that it's safe to do so yet. We can't evaluate the expression until we know everything is fully resolved. For example, if this expression were a subquery that selected the name of a reference from another table, we'd try to eval that here without the query plan being fully resolved and would fail. The calls to `Eval` need to be limited to when we're executing / in our RowIter code.